ir/type.cpp

```diff
@@ -704,12 +751,12 @@ expr SymbolicType::getDummyValue() const {
 
 expr SymbolicType::getTypeConstraints() const {
   expr c(false);
-  c |= isInt()    && i.getTypeConstraints();
-  c |= isFloat()  && f.getTypeConstraints();
-  c |= isPtr()    && p.getTypeConstraints();
-  c |= isArray()  && a.getTypeConstraints();
+  c |= isInt()    && i.getTypeConstraints() && (v.getElementTy().sizeVar() == 0);
+  c |= isFloat()  && f.getTypeConstraints() && (v.getElementTy().sizeVar() == 0);
+  c |= isPtr()    && p.getTypeConstraints() && (v.getElementTy().sizeVar() == 0);
+  c |= isArray()  && a.getTypeConstraints() && (v.getElementTy().sizeVar() == 0);
   c |= isVector() && v.getTypeConstraints();
-  c |= isStruct() && s.getTypeConstraints();
+  c |= isStruct() && s.getTypeConstraints() && (v.getElementTy().sizeVar() == 0);
```

Comment:
symbolic vector brings a new free variable, this set the constraints on it when reasoning non-vector symbolic types. This looks ugly, though. @nunoplopes do you have comments for this?

Reply:
Right, please don't do this :)
There's no need for these constraints. In the worst case we'll finally need the model minimizer to avoid going over equivalent models of types. I had to disable some stuff in pointers as well already. I'll look into implementing that.